ee/tabby-webserver/src/handler.rs

```diff
@@ -72,3 +77,19 @@ async fn distributed_tabby_layer(
 ) -> axum::response::Response {
     ws.worker().dispatch_request(request, next).await
 }
+
+async fn server_setting(
+    State(locator): State<Arc<dyn ServiceLocator>>,
+) -> Result<Json<ServerSetting>, StatusCode> {
+    let setting = match locator.setting().read_security_setting().await {
+        Ok(x) => x,
+        Err(err) => {
+            warn!("Failed to read security setting {}", err);
+            return Err(StatusCode::INTERNAL_SERVER_ERROR);
+        }
+    };
+
+    Ok(Json(ServerSetting {
+        disable_client_side_telemetry: setting.disable_client_side_telemetry,
```

Comment:
If this is to convert the DAO, don't we have a `From` impl for it?

Reply:
mostly not wanna create a practice that whatever in `ServerSetting` should be stored in `ServerSetting` table.

They should be somewhat irrelevant concept, but unfortunately reused the name :/

Reply:
Just concerned about the number of places we might need to duplicate this logic if we won't be using into() - do you mean you want to decouple the DAO and external representation more?